src/lib.rs

```diff
@@ -33,6 +33,8 @@ pub struct Config {
     username: String,
     // Password for signing into a private registry
     password: String,
+    // If a locally pulled version of the container should be used
+    use_local: bool,
```

Comment:
My thought was, instead of introducing a new flag, if we could check the image list and see if the image is present before pulling as default behavior?

And add another flag, `--force` or something, that forces the image to be pulled from a remote registry.

Reply:
If a container image is not available locally, then we would fall back to pull, which makes the UX more pleasant and expected IMO

Reply:
Just a drive by but if I'm reading the use case right, I'd expect us to be able to make reasonable assumptions here without requiring extra configuration knobs. Maybe we're performing unnecessary work if the local image isn't available, and then we'd need to try and handle any remote workflows, but at least the UX would be consistent.

Reply:
With that said, you'd ideally document this behavior so you're covered on all fronts.

Reply:
Gotcha, I went ahead and updated this to have the `--force` flag and default to pulling only if the image is not present locally.=9